app/views/chapters/show.html.erb

```diff
@@ -1,3 +1,12 @@
+<%
+
+  @lesson ||= @chapter.first_lesson
+  @guide ||= @lesson.guide
+  @stats ||= @lesson.stats_for(current_user) if current_user?
```

Comment:
We should avoid placing such logic in views. It is a controller concern.

Reply:
If that is not possible because this view is uses multiple times, then at least we may put in a `before_action` or a `helper`